### PR TITLE
2

### PR DESCRIPTION
**Pull request by TarunRKaushik**

### CODE REVIEW

**Comment by grumpyp** [comment]
Hi, appreciate your PR. There's something missing in our FILE_HANDLERS

can you add that logic?

https://github.com/grumpyp/aixplora/blob/main/backend/utils.py#L4

**Comment by TarunRKaushik** [comment]
Hey, thanks for the message, I've added the load_presentation() call in FILE_HANDLERS.

**Comment by grumpyp** [comment]
Please make sure you test it before opening a PR.

For me it's not working yet.

**Comment by TarunRKaushik** [comment]
so sorry for the delay @grumpyp but I'm facing some issues in building and testing the App. On windows, some of the dependencies aren't supported and on Mac when I run pip install on the requirements.txt, it says some of the dependency versions are conflicting. Should I raise an issue for this?

Anyway I've made the fix for now

**Comment by grumpyp** [comment]
Yes please create an issue, so we can fix it.

File: backend/loaders/ppt_loader.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    text_content = []
+
+    for slide in presentation.slides:
+        slide_text = []
+        for shape in slide.shapes:
+            if shape.has_text_frame:
+                for paragraph in shape.text_frame.paragraphs:
+                    for run in paragraph.runs:
+                        slide_text.append(run.text.strip())
+        text_content.append(slide_text.replace("\n", " "))
```
Comment (grumpyp) [inline backend/loaders/ppt_loader.py:15]: slide_text is type `list`, so, `replace()` is a str builtin.

File: backend/loaders/ppt_loader.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from pptx import Presentation
+import os
+
+def load_presentation(presentation_file: bytes, filename: str) -> str:
```
Comment (grumpyp) [inline backend/loaders/ppt_loader.py:4]: please check the other loaders functions, we added meta data, it's in place there

File: backend/utils.py [hunk 3]
```diff
@@ -8,6 +9,7 @@
 
 FILE_HANDLERS = {
     ".pdf": lambda file: load_pdf(file.file, filename=file.filename, file_meta=file),
+    ".ppt": lambda file: load_presentation(file.file, filename=file.filename),
```
Comment (grumpyp) [inline backend/utils.py:12]: meta

**Comment by grumpyp** [comment]
please make sure to test before requesting a review. I left some comments.

File: backend/loaders/ppt_loader.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            if shape.has_text_frame:
+                for paragraph in shape.text_frame.paragraphs:
+                    for run in paragraph.runs:
+                        slide_text += run.text.strip()
+        text_content += slide_text.replace("\n", " ")
+
+
+    # Write files to misc folder
+    misc_dir = os.path.join(os.getcwd(), "misc")
+    filename = filename.split(".")[0]  #to remove the .ppt extension from the filename
```
Comment (TarunRKaushik) [inline backend/loaders/ppt_loader.py:20]: @grumpyp is this necessary? I had noticed this was done in one of the other loaders but don't see it in the main code now. 
Will the function expect filename without the extension?

Comment (grumpyp) [reply backend/loaders/ppt_loader.py:20]: Please build it like the other loaders. We do not cut any fileextension off there.

**Comment by grumpyp** [comment]
Fixed some stuff #35 merged

**Pull request closed without merging**